extensions/agent/client/src/dialogs/createJobDialog.ts

```diff
@@ -0,0 +1,348 @@
+	private deleteJobConditionDropdown: sqlops.DropDownComponent;
+
+	private model: CreateJobData;
+
+	constructor(ownerUri: string) {
+		this.model = new CreateJobData(ownerUri);
+	}
+
+	public async showDialog() {
+		await this.model.initialize();
+		this.dialog = sqlops.window.modelviewdialog.createDialog(this.DialogTitle);
+		this.generalTab = sqlops.window.modelviewdialog.createTab(this.GeneralTabText);
+		this.stepsTab = sqlops.window.modelviewdialog.createTab(this.StepsTabText);
+		this.alertsTab = sqlops.window.modelviewdialog.createTab(this.AlertsTabText);
+		this.schedulesTab = sqlops.window.modelviewdialog.createTab(this.SchedulesTabText);
+		this.notificationsTab = sqlops.window.modelviewdialog.createTab(this.NotificationsTabText);
+		this.initializeGeneralTab();
+		this.initializeStepsTab();
+		this.initializeAlertsTab();
+		this.initializeSchedulesTab();
+		this.initializeNotificationsTab();
+		this.dialog.content = [this.generalTab, this.stepsTab, this.schedulesTab, this.alertsTab, this.notificationsTab];
+		this.dialog.okButton.onClick(async () => await this.execute());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+		this.dialog.okButton.label = this.OkButtonText;
+		this.dialog.cancelButton.label = this.CancelButtonText;
+		sqlops.window.modelviewdialog.openDialog(this.dialog);
+	}
+
+	private initializeGeneralTab() {
+		this.generalTab.registerContent(async view => {
+			this.nameTextBox = view.modelBuilder.inputBox().component();
+			this.ownerTextBox = view.modelBuilder.inputBox().component();
+			this.categoryDropdown = view.modelBuilder.dropDown().component();
+			this.descriptionTextBox = view.modelBuilder.inputBox().withProperties({
+				multiline: true,
+				height: 200
+			}).component();
+			this.enabledCheckBox = view.modelBuilder.checkBox()
+				.withProperties({
+					label: this.EnabledCheckboxLabel
+				}).component();
+			let formModel = view.modelBuilder.formContainer()
+				.withFormItems([{
+					component: this.nameTextBox,
+					title: this.NameTextBoxLabel
+				}, {
+					component: this.ownerTextBox,
+					title: this.OwnerTextBoxLabel
+				}, {
+					component: this.categoryDropdown,
+					title: this.CategoryDropdownLabel
+				}, {
+					component: this.descriptionTextBox,
+					title: this.DescriptionTextBoxLabel
+				}, {
+					component: this.enabledCheckBox,
+					title: ''
+				}]).withLayout({ width: '100%' }).component();
+
+			await view.initializeModel(formModel);
+
+			this.ownerTextBox.value = this.model.defaultOwner;
+			this.categoryDropdown.values = this.model.jobCategories;
+			this.categoryDropdown.value = this.model.jobCategories[0];
+			this.enabledCheckBox.checked = this.model.enabled;
+			this.descriptionTextBox.value = '';
+		});
+	}
+
+	private initializeStepsTab() {
+		this.stepsTab.registerContent(async view => {
+			this.stepsTable = view.modelBuilder.table()
+				.withProperties({
+					columns: [
+						this.StepsTable_StepColumnString,
```

Comment:
I don't think we use `_` character in variable names in this project.  Probably we should follow normal naming conventions for these fields unless there is something special about them.